File: tests/benchmarks/macro/doors.py
def simulate():
    doors = [False] * 100
    for p in range(1, 101):
        for idx in range(p - 1, 100, p):
            doors[idx] = not doors[idx]
    return sum(1 for d in doors if d)
total = 0
for _ in range(200):
    total += simulate()
print(total)

File: tests/benchmarks/macro/doors.lua
-- 100 doors simulation. Equal-work port of doors.ori / doors.py.
-- Run under LuaJIT with -joff (strict-interpreter north-star comparator).
local function simulate()
  local doors = {}
  for i = 1, 100 do doors[i] = false end
  for p = 1, 100 do
    for idx = p, 100, p do
      doors[idx] = not doors[idx]
    end
  end
  local open = 0
  for i = 1, 100 do
    if doors[i] then open = open + 1 end
  end
  return open
end

local total = 0
for _ = 1, 200 do
  total = total + simulate()
end
print(total)
